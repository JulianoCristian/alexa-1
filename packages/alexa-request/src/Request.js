import { Request as Constants } from 'alexa-constants';

export default class Request {
  static session = (...args) => new Request().session(...args);
  static intent = (...args) => new Request().intent(...args);
  static launchRequest = (...args) => new Request().launchRequest(...args);
  static sessionEndedRequest = (...args) => new Request().sessionEndedRequest(...args);

  constructor(state = {}) {
    this.state = state;
  }

  session(session) {
    return new Request({
      ...this.state,
      session: {
        ...this.state.session,
        ...session
      }
    });
  }

  intent(name, slots = {}) {
    const slotData = Object.entries(slots).reduce((state, [name, value]) => ({ ...state, [name]: { name, value } }), {});
    return new Request({
      ...this.state,
      request: {
        type: Constants.Intent,
        intent: {
          name: name,
          ...(Object.keys(slotData).length ? { slots: slotData } : null)
        }
      }
    });
  }

  launchRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.Launch
      }
    });
  }

  sessionEndedRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.SessionEnded
      }
    });
  }

  build() {
    return { ...this.state };
  }
}
