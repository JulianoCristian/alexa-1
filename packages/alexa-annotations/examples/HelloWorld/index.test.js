import Request from 'alexa-request';
import HelloWorld from './index';

const DefaultRequest = Request.session({
  application: { applicationId: 'my-app-id' }
});

test('LaunchRequest', () => {
  const event = DefaultRequest.launchRequest().build();
  return expect(HelloWorld(event)).resolves.toMatchSnapshot();
});

test('HelloWorldIntent', () => {
  const event = DefaultRequest.intent('HelloWorldIntent').build();
  return expect(HelloWorld(event)).resolves.toMatchSnapshot();
});

test('AMAZON.HelpIntent', () => {
  const event = DefaultRequest.intent('AMAZON.HelpIntent').build();
  return expect(HelloWorld(event)).resolves.toMatchSnapshot();
});

test('Credits', () => {
  const event = DefaultRequest.intent('Credits').build();
  return expect(HelloWorld(event)).resolves.toMatchSnapshot();
});
