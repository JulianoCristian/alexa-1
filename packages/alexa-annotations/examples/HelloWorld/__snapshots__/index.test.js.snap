// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMAZON.HelpIntent 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "You can say hello to me!",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "You can say hello to me!",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;

exports[`Credits 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Credits: Cameron Hunter <hello@cameronhunter.co.uk> (http://cameronhunter.co.uk)",
      "title": "Hello World",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "text": "Hello World was created by Cameron Hunter",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`HelloWorldIntent 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Hello World!",
      "title": "Greeter",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "text": "Hello World!",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`LaunchRequest 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "Welcome to the Alexa Skills Kit, you can say hello",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "You can say hello",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;
